### PR TITLE
Handle null extent

### PR DESCRIPTION
**Pull request by GretaCB**
Per https://github.com/mapbox/tilelive.js/issues/118

cc @rclark

### CODE REVIEW

File: test/stream-pyramid.test.js [hunk 1]
```diff
@@ -172,3 +172,26 @@ test('pyramid: err + retry', function(assert) {
...
+            });
+        }
+    };
+
+    require('../lib/stream-util').retryBackoff = 1;
+    var get = tilelive.createReadStream(fakesrc, {type:'pyramid'});
+    var put = tilelive.createWriteStream(new Timedsource({}));
+    get.on('error', function(err) {         
+        assert.equal(err.message, 'Invalid extent value');
+        assert.end();});
```
Comment (rclark) [inline test/stream-pyramid.test.js]: if you `assert.plan` you don't need to `assert.end`

File: lib/stream-pyramid.js [hunk 2]
```diff
@@ -65,6 +65,12 @@ Pyramid.prototype._params = function(callback) {
 
         stream.bboxes = {};
         var bounds = limitBounds(stream.bounds);
+
+        // Validate bounds values
+        for (var b = 0; b < bounds.length; b++) {
+            if (isNaN(bounds[b])) return stream.emit('error', new Error('Invalid extent value'));
```
Comment (rclark) [inline lib/stream-pyramid.js]: `isNaN(null) === false`. You may need to get a little trickier in your condition that the value is numeric. This isn't a one-liner in javascript unfortunately...

**Comment by GretaCB** [comment]
Added more strict `number` centric validation. 

Also, just fyi, looks like the [limitBounds function](https://github.com/mapbox/tilelive.js/blob/master/lib/stream-util.js#L228-L234) is a bit forgiving. For example, my extent test includes the following bounds:

```
[null, 128379137, NaN, undefined]
```

And after running through limitbounds, these values become:

```
[0, 90, NaN, NaN]
```

**Comment by yhahn** [comment]
Hrm, could we use and/or consolidate this logic with https://github.com/mapbox/tilelive.js/blob/master/lib/tilelive.js#L234-L249 ?

**Comment by rclark** [comment]
As I see it the bug we're trying to fix here is that `NaN` in source bounds put the read streams in a loop that never ends.

Having `tilelive.verify` reject these would be useful in many contexts, but copy streams don't explicitly run `.verify` on the source that you hand them. Maybe we should change that?

**Comment by yhahn** [comment]
@rclark @GretaCB right, for a more conservative start to this concept you could pass an object with `bounds` (and any other params you care about for copying, minzoom/maxzoom?) only to `tilelive.verify`.

**Pull request merged**